### PR TITLE
common: Introduce struct Level to Option

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

**Comment by liewegas** [comment]
I have no opinion on the syntactic construction we use here.  @tchaikov ?

File: src/common/options.h [hunk 1]
```diff
@@ -39,17 +39,21 @@ struct Option {
    * Advanced: for users, configures some internal behaviour
    * Development: not for users.  May be dangerous, may not be documented.
    */
-  enum level_t {
-    LEVEL_BASIC,
-    LEVEL_ADVANCED,
-    LEVEL_DEV,
+  struct Level {
```
Comment (tchaikov) [inline src/common/options.h]: if you are interested a syntax like this, please consider http://en.cppreference.com/w/cpp/language/enum#Scoped_enumerations.

Comment (shinobu-x) [reply src/common/options.h]: nice...

**Comment by shinobu-x** [comment]
@tchaikov like now?

File: src/common/options.h [hunk 2]
```diff
@@ -39,17 +39,19 @@ struct Option {
    * Advanced: for users, configures some internal behaviour
    * Development: not for users.  May be dangerous, may not be documented.
    */
-  enum level_t {
-    LEVEL_BASIC,
-    LEVEL_ADVANCED,
-    LEVEL_DEV,
+  enum struct Level{
```
Comment (tchaikov) [inline src/common/options.h]: please add a space after `Level`.

File: src/common/options.h [hunk 3]
```diff
@@ -39,17 +39,19 @@ struct Option {
...
   };
 
-  const char *level_to_str(level_t l) const {
-    switch(l) {
-      case LEVEL_BASIC: return "basic";
-      case LEVEL_ADVANCED: return "advanced";
-      case LEVEL_DEV: return "developer";
+  typedef Level level_t;
+
+  const char* level_to_str(level_t t) const {
```
Comment (tchaikov) [inline src/common/options.h]: why do you think `t` is a better name?

Comment (shinobu-x) [reply src/common/options.h]: @tchaikov no idea :)

Comment (shinobu-x) [reply src/common/options.h]: @tchaikov that may probably be heritage of `Type`

**Comment by tchaikov** [comment]
it looks good to me. but i am not a big fan of this kind of syntax change en mass. if @jcsp is fine with it, so am i.

**Pull request closed without merging**